File: WebClient/src/app/pages/invention-container/invention-categories/shared/invention-category.factory.ts
import { Injectable } from '@angular/core';
import {InventionCategory} from './invention-category.model';
import {CrudModelFactory} from '../../../../shared/factories/crud-model.factory';

@Injectable({
  providedIn: 'root'
})
export class InventionCategoryFactory extends CrudModelFactory<InventionCategory>{
  create(jsonData: any): InventionCategory {
    return jsonData ? new InventionCategory(
      jsonData.id,
      jsonData.code,
      jsonData.name,
      jsonData.description
    ) : new InventionCategory();
  }
}

File: Application/Factories/InventionCategoryFactory.cs
using Application.Contracts.Factories;
using Application.Dtos;
using AutoMapper;
using Domain.Aggregates;

namespace Application.Factories
{
    public class InventionCategoryFactory : Factory<InventionCategoryDto, InventionCategory>,  IInventionCategoryFactory
    {
        public InventionCategoryFactory(IMapper mapper) : base(mapper)
        {
        }

        public override InventionCategory Create(InventionCategoryDto dto)
        {
            return new InventionCategory(
                dto.Code,
                dto.Name,
                dto.Description
            );
        }
    }
}